compiler/noirc_frontend/src/elaborator/patterns.rs

```diff
@@ -791,7 +791,7 @@ impl Elaborator<'_> {
         typ
     }
 
-    fn instantiate(
+    pub(super) fn instantiate(
```

Comment:
Is this change needed?

Reply:
Ah, not any longer. An earlier version of this PR shared more code which I later figured out could be replaced with `type_check_variable`.